filesystem/flysystem/FlysystemAssetStore.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace SilverStripe\Filesystem\Flysystem;
+
+use Config;
+use Injector;
+use InvalidArgumentException;
+use League\Flysystem\Exception;
+use League\Flysystem\Filesystem;
+use SilverStripe\Filesystem\Storage\AssetNameGenerator;
+use SilverStripe\Filesystem\Storage\AssetStore;
+
+/**
+ * Asset store based on flysystem Filesystem as a backend
+ */
+class FlysystemAssetStore implements AssetStore {
+
+	/**
+	 * @var Filesystem
+	 */
+	protected $filesystem;
```

Comment:
I thought you were against private. :) https://github.com/silverstripe/silverstripe-framework/pull/3874